### PR TITLE
[Merged by Bors] - feat: port Order.Disjoint

### PR DESCRIPTION
**Pull request by hrmacbeth**
sha 1fc36cc9c8264e6e81253f88be7fb2cb6c92d76a

### CODE REVIEW

File: Mathlib/Order/Disjoint.lean [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+
+/-- Two elements `x` and `y` are complements of each other if `x ⊔ y = ⊤` and `x ⊓ y = ⊥`. -/
+structure IsCompl [PartialOrder α] [BoundedOrder α] (x y : α) : Prop where
+  /-- If `x` and `y` are to be complementary in an order, they should be disjoint. -/
+  protected Disjoint : Disjoint x y
+  /-- If `x` and `y` are to be complementary in an order, they should be codisjoint. -/
+  protected Codisjoint : Codisjoint x y
+#align is_compl IsCompl
+
+theorem is_compl_iff [PartialOrder α] [BoundedOrder α] {a b : α} :
```
Comment (semorrison) [inline Mathlib/Order/Disjoint.lean:432]:
```suggestion
theorem isCompl_iff [PartialOrder α] [BoundedOrder α] {a b : α} :
```
Will need an #align to go with it.

File: Mathlib/Order/Disjoint.lean [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+  id
+#align disjoint.dual Disjoint.dual
+
+theorem Codisjoint.dual [SemilatticeSup α] [OrderTop α] {a b : α} :
+    Codisjoint a b → Disjoint (toDual a) (toDual b) :=
+  id
+#align codisjoint.dual Codisjoint.dual
+
+@[simp]
+theorem disjoint_to_dual_iff [SemilatticeSup α] [OrderTop α] {a b : α} :
```
Comment (semorrison) [inline Mathlib/Order/Disjoint.lean:392]:
```suggestion
theorem disjoint_toDual_iff [SemilatticeSup α] [OrderTop α] {a b : α} :
```
Also below. These will need #align. (I'd recommend leaving in all #aligns, even when mathport gets the names right; easier when you have to put one back in later!)

File: Mathlib/Order/Disjoint.lean [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+protected theorem codisjoint_iff [OrderTop α] [OrderTop β] {x y : α × β} :
+    Codisjoint x y ↔ Codisjoint x.1 y.1 ∧ Codisjoint x.2 y.2 :=
+  @Prod.disjoint_iff αᵒᵈ βᵒᵈ _ _ _ _ _ _
+#align prod.codisjoint_iff Prod.codisjoint_iff
+
+protected theorem is_compl_iff [BoundedOrder α] [BoundedOrder β] {x y : α × β} :
+    IsCompl x y ↔ IsCompl x.1 y.1 ∧ IsCompl x.2 y.2 := by
+  simp_rw [is_compl_iff, Prod.disjoint_iff, Prod.codisjoint_iff, and_and_and_comm]
+  rfl
+#align prod.is_compl_iff Prod.is_compl_iff
```
Comment (semorrison) [inline Mathlib/Order/Disjoint.lean:579]:
```suggestion
protected theorem isCompl_iff [BoundedOrder α] [BoundedOrder β] {x y : α × β} :
    IsCompl x y ↔ IsCompl x.1 y.1 ∧ IsCompl x.2 y.2 := by
  simp_rw [is_compl_iff, Prod.disjoint_iff, Prod.codisjoint_iff, and_and_and_comm]
  rfl
#align prod.is_compl_iff Prod.isCompl_iff
```

Comment (semorrison) [reply Mathlib/Order/Disjoint.lean:579]: Several more below in the next section.

File: Mathlib/Order/Disjoint.lean [hunk 4]
```diff
@@ -0,0 +1,635 @@
...
+end
+
+/-- A complemented bounded lattice is one where every element has a (not necessarily unique)
+complement. -/
+class ComplementedLattice (α) [Lattice α] [BoundedOrder α] : Prop where
+  /-- In a `ComplementedLattice`, every element admits a complement. -/
+  exists_is_compl : ∀ a : α, ∃ b : α, IsCompl a b
+#align complemented_lattice ComplementedLattice
+
+export ComplementedLattice (exists_is_compl)
```
Comment (semorrison) [inline Mathlib/Order/Disjoint.lean:622]:
```suggestion
  exists_isCompl : ∀ a : α, ∃ b : α, IsCompl a b
#align complemented_lattice ComplementedLattice

export ComplementedLattice (exists_isCompl)
```

File: Mathlib/Order/Disjoint.lean [hunk 5]
```diff
@@ -0,0 +1,635 @@
...
+
+namespace ComplementedLattice
+
+variable [Lattice α] [BoundedOrder α] [ComplementedLattice α]
+
+instance : ComplementedLattice αᵒᵈ :=
+  ⟨fun a ↦
+    let ⟨b, hb⟩ := exists_is_compl (show α from a)
```
Comment (semorrison) [inline Mathlib/Order/Disjoint.lean:630]:
```suggestion
    let ⟨b, hb⟩ := exists_isCompl (show α from a)
```

File: Mathlib/Order/Disjoint.lean [hunk 6]
```diff
@@ -0,0 +1,650 @@
...
+
+/-- Two elements `x` and `y` are complements of each other if `x ⊔ y = ⊤` and `x ⊓ y = ⊥`. -/
+structure IsCompl [PartialOrder α] [BoundedOrder α] (x y : α) : Prop where
+  /-- If `x` and `y` are to be complementary in an order, they should be disjoint. -/
+  protected Disjoint : Disjoint x y
+  /-- If `x` and `y` are to be complementary in an order, they should be codisjoint. -/
+  protected Codisjoint : Codisjoint x y
+#align is_compl IsCompl
+
+theorem IsCompl_iff [PartialOrder α] [BoundedOrder α] {a b : α} :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Disjoint.lean:438]: Shouldn't the theorems all have `isCompl` rather than `IsCompl`?

Comment (semorrison) [reply Mathlib/Order/Disjoint.lean:438]: Yes.

Comment (hrmacbeth) [reply Mathlib/Order/Disjoint.lean]: Thanks for noticing, I hope I have maintained the correct distinction in the latest commit.

**Comment by semorrison** [comment]
Looks good, thanks!

bors merge

**Pull request closed without merging**